remill/BC/Util.cpp

```diff
@@ -1823,7 +1825,7 @@ BuildIndexes(const llvm::DataLayout &dl, llvm::Type *type, size_t offset,
       }
     }
 
-  } else if (auto seq_type = llvm::dyn_cast<llvm::SequentialType>(type);
+  } else if (auto seq_type = llvm::dyn_cast<llvm::ArrayType>(type);
```

Comment:
You have strengthened the check here; will this work for the other prior derivations of `SequentialType`?

Reply:
I added a case handling fixed and variable vector types and emitting a `FATAL` if that case ever hits. Ideally it should not, but in case of some optimization makes it hit, we'll know if it needs to be handled.